backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Core/Services/SingleRightService.cs

```diff
@@ -0,0 +1,37 @@
+using Altinn.AccessManagement.UI.Core.ClientInterfaces;
+using Altinn.AccessManagement.UI.Core.ClientInterfaces.MockClientInterfaces;
+using Altinn.AccessManagement.UI.Core.Models.SingleRight.CheckDelegationAccess;
+using Altinn.AccessManagement.UI.Core.Models.SingleRight.CheckDelegationAccess.CheckDelegationAccessDto;
+using Altinn.AccessManagement.UI.Core.Services.Interfaces;
+
+namespace Altinn.AccessManagement.UI.Core.Services
+{
+    /// <inheritdoc />
+    public class SingleRightService : ISingleRightService
+    {
+        private readonly ISingleRightClient _singleRightClient;
+        private readonly ISingleRightMockClient _singleRightMockClient;
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="SingleRightMockClient" /> class
+        /// </summary>
+        public SingleRightService(ISingleRightClient singleRightClient, ISingleRightMockClient singleRightMockClient)
+        {
+            _singleRightClient = singleRightClient;
+            _singleRightMockClient = singleRightMockClient;
+        }
+
+        /// <inheritdoc />
+        public async Task<List<DelegationAccessCheckResponse>> CheckDelegationAccess(string partyId, CheckDelegationAccessDto request)
+        {
+            bool isMock = true;
+
+            if (isMock)
+            {
+                return _singleRightMockClient.UserDelegationAccessCheck(partyId, request);
+            }
```

Comment:
After merge with the new Mock setup in https://github.com/Altinn/altinn-access-management-frontend/pull/448
this can be moved to the UseMock setup in program.cs

Reply:
Planned to do so when #448 was merged in, that's why I didn't add any reviewers yet